PerimeterXModule/PxModule.cs

```diff
@@ -1,629 +1,671 @@
+				PostPageRequestedActivity(pxContext);
+			}
+		}
+
+		private static string ByteArrayToHexString(byte[] input)
+		{
+			StringBuilder sb = new StringBuilder(input.Length * 2);
+			foreach (byte b in input)
+			{
+				sb.Append(PxConstants.HEX_ALPHABET[b >> 4]);
+				sb.Append(PxConstants.HEX_ALPHABET[b & 0xF]);
+			}
+			return sb.ToString();
+		}
+
+		private void HandleVerification(HttpApplication application)
+		{
+			PxModuleConfigurationSection config = (PxModuleConfigurationSection)ConfigurationManager.GetSection(PxConstants.CONFIG_SECTION);
+			bool verified = blockingScore > pxContext.Score;
+
+			PxLoggingUtils.LogDebug(string.Format("Request score: {0}, blocking score: {1}, monitor mode status: {2}.", pxContext.Score, blockingScore, config.MonitorMode == true ? "On" : "Off"));
+
+			if (verified)
+			{
+				if (config.MonitorMode)
+				{
+					PxLoggingUtils.LogDebug("Monitor Mode is activated. passing request");
+				}
+
+				HandleCredentialsIntelligence(application, config);
+
+				PxLoggingUtils.LogDebug(string.Format("Valid request to {0}", application.Context.Request.RawUrl));
+				PostPageRequestedActivity(pxContext);
+			}
+			else
+			{
+				PxLoggingUtils.LogDebug(string.Format("Invalid request to {0}", application.Context.Request.RawUrl));
+				PostBlockActivity(pxContext, config);
+			}
+
+			SetPxhdAndVid(pxContext);
+			// If implemented, run the customVerificationHandler.
+			if (customVerificationHandlerInstance != null)
+			{
+				customVerificationHandlerInstance.Handle(application, pxContext, config);
+			}
+			// No custom verification handler -> continue regular flow
+			else if (!verified && !pxContext.MonitorRequest)
+			{
+				BlockRequest(pxContext, config);
+				application.CompleteRequest();
+			}
+		}
+
+		private static void SetPxhdAndVid(PxContext pxContext)
+		{
+
+			if (!string.IsNullOrEmpty(pxContext.Pxhd))
+			{
+				string pxhd = PxConstants.COOKIE_PXHD_PREFIX + "=" + pxContext.Pxhd + "; path=/";
+				pxContext.ApplicationContext.Response.AddHeader("Set-Cookie", pxhd);
+			}
+		}
+
+		private void HandleCredentialsIntelligence(HttpApplication application, PxModuleConfigurationSection config)
+		{
+			if (config.LoginCredentialsExtractionEnabled && pxContext.LoginCredentialsFields != null)
+			{
+				if (pxContext.IsBreachedAccount())
+				{
+					application.Context.Request.Headers.Add(config.CompromisedCredentialsHeader, JSON.SerializeDynamic(pxContext.Pxde.breached_account, PxConstants.JSON_OPTIONS));
+				}
+
+				if (config.AdditionalS2SActivityHeaderEnabled)
+				{
+					Activity activityPayload = AdditionalS2SUtils.CreateAdditionalS2SActivity(config, null, null, pxContext);
+					application.Context.Request.Headers.Add("px-additional_activity", JSON.SerializeDynamic(activityPayload, PxConstants.JSON_OPTIONS));
+					application.Context.Request.Headers.Add("px-backend-activity-url", PxConstants.FormatBaseUri(config) + PxConstants.ACTIVITIES_API_PATH);

```

Comment:
The header names don't match the spec - https://github.com/PerimeterX/px-enforcer-spec/wiki/Credentials-Intelligence#sending-the-additional-s2s-manually-via-request-header